src/bridge/test-metrics.c

```diff
@@ -850,6 +850,58 @@ test_cpu_cores (void)
   g_object_unref (transport);
 }
 
+static void
+test_cpu_temperature (void)
+{
+  MockTransport *transport = mock_transport_new ();
+
+  g_autoptr (JsonObject) options = json_obj ("{ 'metrics': [ { 'name': 'cpu.temperature' } ],"
+                                  "  'interval': 1000"
+                                  "}");
+
+  g_signal_connect (transport, "closed", G_CALLBACK (on_transport_closed), NULL);
```

Comment:
It's very strange that this signal handler is between the site where the JSON is parsed and where it's used.  Please move it up or down.